From 9670d9e5eb7acb9efb0f2f1a798a09baace53d82 Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Mon, 2 Jun 2025 09:56:57 +0800
Subject: FFmpeg: 7.2 compile fix

AV_CODEC_CAP_SUBFRAMES was deprecated without replacement and removed.

Pick-to: 6.8 6.5
Change-Id: I45ac1d0d603f88701bdb727a7a472cfea8f32d5d
Reviewed-by: Pavel Dubsky <pavel.dubsky@qt.io>
Reviewed-by: Tim Blechmann <tim.blechmann@qt.io>
(cherry picked from commit 65b11b9f97632101041134e04713697b17819884)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit ef866d101e58832f0bf86150ed8328e2d3ec9c49)

Upstream: https://code.qt.io/cgit/qt/qtmultimedia.git/commit/?h=6.9.2&id=9670d9e5eb7acb9efb0f2f1a798a09baace53d82

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 src/plugins/multimedia/ffmpeg/qffmpegcodecstorage.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plugins/multimedia/ffmpeg/qffmpegcodecstorage.cpp b/src/plugins/multimedia/ffmpeg/qffmpegcodecstorage.cpp
index c154fd835..360a33909 100644
--- a/src/plugins/multimedia/ffmpeg/qffmpegcodecstorage.cpp
+++ b/src/plugins/multimedia/ffmpeg/qffmpegcodecstorage.cpp
@@ -99,7 +99,9 @@ void dumpCodecInfo(const Codec &codec)
         { AV_CODEC_CAP_DR1, "DRAW_HORIZ_DR1" },
         { AV_CODEC_CAP_DELAY, "DELAY" },
         { AV_CODEC_CAP_SMALL_LAST_FRAME, "SMALL_LAST_FRAME" },
+#ifdef AV_CODEC_CAP_SUBFRAMES
         { AV_CODEC_CAP_SUBFRAMES, "SUBFRAMES" },
+#endif
         { AV_CODEC_CAP_EXPERIMENTAL, "EXPERIMENTAL" },
         { AV_CODEC_CAP_CHANNEL_CONF, "CHANNEL_CONF" },
         { AV_CODEC_CAP_FRAME_THREADS, "FRAME_THREADS" },
-- 
cgit v1.2.3

